go/master/client.go

```diff
@@ -20,6 +25,51 @@ type Client struct {
 	ch   chan []byte
 }
 
+// MasterAddresser provide master address
+type masterAddresser struct {
+	client    *clientv3.Client
+	timeout   time.Duration
+	endpoints []string
+}
+
+// Address return the address
+func (m masterAddresser) Address() string {
+	for {
+		ctx, cancel := context.WithTimeout(context.Background(), m.timeout)
+		resp, err := m.client.Get(ctx, masterAddrPath)
```

Comment:
Clients should use `watch` to retrieve notification about master node events. `masterMonitor` should also change to handle these events.